### PR TITLE
leveldb: add write delay statistic

### PR DESCRIPTION
**Pull request by rjl493456442**
Implements #181

### CODE REVIEW

**Comment by rjl493456442** [comment]
@syndtr  PTAL :)

File: leveldb/db.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ type DB struct {
 
 	// Stats.
 	aliveSnaps, aliveIters int32
+	writeDelayStat         *wDelayStat
```
Comment (syndtr) [inline leveldb/db.go]: I think we can just unpack `wDelayStat` struct into DB struct. Also I think using `sync/atomic` is enough (yes, inconsistency between counter and timer might happen but I think that's not a big issue in this case), you can cast `time.Duration` to `int64`.

**Review by syndtr — CHANGES_REQUESTED** [review]

**Review by syndtr — APPROVED** [review]

**Review by syndtr — CHANGES_REQUESTED** [review]
I think we can just unpack `wDelayStat` struct into DB struct. Also I think using `sync/atomic` is enough (yes, inconsistency between counter and timer might happen but I think that's not a big issue in this case), you can cast `time.Duration` to `int64`.

**Comment by rjl493456442** [comment]
@syndtr Updated!

**Comment by syndtr** [comment]
Merged at 34011bf325bce385408353a30b101fe5e923eb6e. Thanks!

**Pull request closed without merging**